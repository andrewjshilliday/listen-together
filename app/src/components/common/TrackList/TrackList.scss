.track {
  width: 100%;
  display: flex;
}

.track-option {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 50px;
  width: 50px;
}
