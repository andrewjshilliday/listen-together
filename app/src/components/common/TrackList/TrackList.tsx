import React from 'react';
import { useDispatch } from 'react-redux';
import moment from 'moment';
import { MusicKitService } from '../../../services';
import { addToPlaylist } from '../../../store/room';
import styled from 'styled-components';

interface TrackListProps {
  tracks: MusicKit.MediaItem[],
  showArtwork?: boolean,
  showArtist?: boolean,
  showAlbum?: boolean
}

const TrackList: React.FC<TrackListProps> = ({ tracks, showArtwork, showArtist, showAlbum }) => {
  const dispatch = useDispatch();

  return (
    <TrackListContainer>
      {tracks.map((track: MusicKit.MediaItem) => (
        <Track key={track.id}>
          <TrackOption onClick={() => dispatch(addToPlaylist(track))}><i className="fas fa-plus"></i></TrackOption>
          {showArtwork && <img src={MusicKitService.FormatArtwork(track.attributes.artwork, 44)} className="img-fluid rounded" alt={track.attributes.name} />}
          <TrackName artist={showArtist} album={showAlbum} className="text-truncate">{track.attributes.name}</TrackName>
          {showArtist && <TrackArtist className="text-truncate">{track.attributes.artistName}</TrackArtist>}
          {showAlbum && <TrackAlbum className="track-album text-truncate">{track.attributes.albumName}</TrackAlbum>}
          <TrackDuration className="track-duration">{moment.utc(track.attributes.durationInMillis).format('m:ss')}</TrackDuration>
        </Track>
      ))}
    </TrackListContainer>
  );
}

export default TrackList;


const TrackListContainer = styled.div``;
const Track = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  height: 50px;
  border-radius: .5em;
  padding-right: 15px;
  &:hover { background-color: var(--background-darker); }
`;
const TrackOption = styled.div`
  display: flex;
  justify-content: center;
  width: 50px;
  &:hover {
    cursor: pointer;
    color: var(--primary);
  }
`;
const TrackName = styled.span<{artist?: boolean, album?: boolean}>`
  ${props => props.artist && props.album && `
    max-width: calc(100% - 450px);
    width: 50%;
  `}
  ${props => props.artist && !props.album && `
    max-width: calc(100% - 280px);
    width: 75%;
  `}
  ${props => !props.artist && props.album && `
    max-width: calc(100% - 280px);
    width: 75%;
  `}
  ${props => !props.artist && !props.album && `
    max-width: calc(100% - 110px);
    width: calc(100% - 110px);
  `}
`;
const TrackArtist = styled.span`
  width: 25%;
  max-width: 170px;
  min-width: 90px;
`;
const TrackAlbum = styled.span`
  width: 25%;
  max-width: 170px;
  min-width: 90px;
`;
const TrackDuration = styled.span`
  width: 60px;
  text-align: right;
`;