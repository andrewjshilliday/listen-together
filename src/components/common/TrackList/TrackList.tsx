import React from 'react';
import { useRoom } from '../../providers';
import { FormatArtwork } from '../../../services/MusicKit';
import './TrackList.scss';

interface TrackListProps {
  tracks: any,
  displayArtwork: boolean
}

const TrackList: React.FC<TrackListProps> = ({ tracks, displayArtwork }) => {
  const roomProvider = useRoom();

  const addToPlaylist = (track: any) => {
    roomProvider.addToPlaylist(track);
  };

  return (
    <div className="track-list">
      {tracks.map((track: any) => (
        <div key={track.id} className="track">
          <div className="track-option" onClick={() => addToPlaylist(track)}><i className="fas fa-plus"></i></div>
          {displayArtwork && <img src={FormatArtwork(track.attributes.artwork, 44)} className="img-fluid rounded" alt={track.attributes.name} />}
          <p>{track.attributes.name}</p>
        </div>
      ))}
    </div>
  );
}

export default TrackList;
