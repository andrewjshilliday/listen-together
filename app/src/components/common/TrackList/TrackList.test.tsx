import React from 'react';
import renderer from 'react-test-renderer';
import TrackList from './TrackList';

jest.mock('../../../services/MusicKit');
jest.mock('../../providers/RoomProvider', () => require('../../../__mocks__/providers/RoomProvider.mock').default);

it('renders correctly', () => {
  const tree = renderer.create(<TrackList tracks={require('../../../__mocks__/objects/Tracks.mock').default} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with artwork, artist and album', () => {
  const tree = renderer.create(<TrackList tracks={require('../../../__mocks__/objects/Tracks.mock').default} showArtwork={true} showArtist={true} showAlbum={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
