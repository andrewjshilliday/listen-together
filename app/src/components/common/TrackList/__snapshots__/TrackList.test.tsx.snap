// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:41
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      What About Us
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:30
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      I Got My Eyes on You (Locked & Loaded)
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:11
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      I Walk Alone
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:44
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Feelin' Like a Million
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:34
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Gotta Get Into Something
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:04
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Got to Get Up
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      2:37
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Feed the Babies
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:46
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Pearl Cadillac
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:05
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      When I'm Gone
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:48
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      The Guitar Man
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:26
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Low Down Rolling Stone
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:18
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      The Governor
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      2:21
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Don't Wait Til Tomorrow
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:05
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Dirty Dishes Blues
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:03
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Highway 71 (Bonus Track)
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:31
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <span
      className="sc-AxhCb bDqOP text-truncate"
    >
      Did Dat (Bonus Track)
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      6:42
    </span>
  </div>
</div>
`;

exports[`renders correctly with artwork, artist and album 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="This Land"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:41
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="What About Us"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      What About Us
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:30
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="I Got My Eyes on You (Locked & Loaded)"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      I Got My Eyes on You (Locked & Loaded)
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:11
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="I Walk Alone"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      I Walk Alone
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:44
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Feelin' Like a Million"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Feelin' Like a Million
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:34
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Gotta Get Into Something"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Gotta Get Into Something
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:04
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Got to Get Up"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Got to Get Up
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      2:37
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Feed the Babies"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Feed the Babies
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:46
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Pearl Cadillac"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Pearl Cadillac
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:05
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="When I'm Gone"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      When I'm Gone
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:48
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="The Guitar Man"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      The Guitar Man
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:26
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Low Down Rolling Stone"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Low Down Rolling Stone
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:18
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="The Governor"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      The Governor
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      2:21
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Don't Wait Til Tomorrow"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Don't Wait Til Tomorrow
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      4:05
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Dirty Dishes Blues"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Dirty Dishes Blues
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      5:03
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Highway 71 (Bonus Track)"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Highway 71 (Bonus Track)
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      3:31
    </span>
  </div>
  <div
    className="sc-AxirZ iZnMCv"
  >
    <div
      className="sc-AxiKw jNumRE"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </div>
    <img
      alt="Did Dat (Bonus Track)"
      className="img-fluid rounded"
    />
    <span
      className="sc-AxhCb iZNYZP text-truncate"
    >
      Did Dat (Bonus Track)
    </span>
    <span
      className="sc-AxhUy jSOFHT text-truncate"
    >
      Gary Clark Jr.
    </span>
    <span
      className="sc-AxgMl bflOSJ track-album text-truncate"
    >
      This Land
    </span>
    <span
      className="sc-AxheI TytHM track-duration"
    >
      6:42
    </span>
  </div>
</div>
`;
